### PR TITLE
Deduplicate and simplify code.

### PR DESCRIPTION
**Pull request by matz-e**
As suggested in #95.

Apparently, I'm unable to fix the LLVM compilation. Any ideas?

### CODE REVIEW

**Comment by wizmer** [comment]
I think there is a constness issue.
It wants to instantiate `'morphio::maxDistanceToCenterOfGravity<gsl::span<const std::array<float, 3> > >`
although you explicitely instantiated:
template float maxDistanceToCenterOfGravity(const range<const Point>& points);

File: src/mut/soma.cpp [hunk 1]
```diff
@@ -40,19 +40,7 @@ float Soma::surface() const
...
-    const auto c = center();
-    float dist = distance(c, ps[0]);
-    for (size_t i = 1; i < ps.size(); ++i) {
-        const float d = distance(c, ps[i]);
-        if (d > dist) {
-            dist = d;
-        }
-    }
-    return dist;
+    return maxDistanceToCenterOfGravity(points());
```
Comment (wizmer) [inline src/mut/soma.cpp]:
```suggestion
    return maxDistanceToCenterOfGravity(_properties->_somaLevel._points);
```

Comment (wizmer) [reply src/mut/soma.cpp]: But `points()` should already be a `const range` so I don't understand.

File: src/mut/soma.cpp [hunk 2]
```diff
@@ -40,19 +40,7 @@ float Soma::surface() const
...
-    const auto c = center();
-    float dist = distance(c, ps[0]);
-    for (size_t i = 1; i < ps.size(); ++i) {
-        const float d = distance(c, ps[i]);
-        if (d > dist) {
-            dist = d;
-        }
-    }
-    return dist;
+    return maxDistanceToCenterOfGravity(_properties->_somaLevel._points);
```
Comment (wizmer) [inline src/mut/soma.cpp]: Sorry, I wanted to suggest on the other one (the non mutable)

Comment (matz-e) [reply src/mut/soma.cpp]: Just got that.

**Pull request merged**